safe/gui/tools/print_report_dialog.py

```diff
@@ -434,20 +571,37 @@ def template_chooser_clicked(self):
         # noinspection PyCallByClass,PyTypeChecker
         file_name = QtGui.QFileDialog.getOpenFileName(
             self,
-            self.tr('Select report'),
+            tr('Select report'),
             '',
-            self.tr('QGIS composer templates (*.qpt *.QPT)'))
+            tr('QGIS composer templates (*.qpt *.QPT)'))
         self.template_path.setText(file_name)
 
-    def toggle_template_selectors(self, checked):
-        if checked:
+    def toggle_template_selector(self):
+        """Auto-connect slot for template selector elements behaviour.
+
+        .. versionadded: 4.3.0
+        """
+        if self.search_directory_radio.isChecked():
             self.template_combo.setEnabled(True)
-            self.template_path.setEnabled(False)
-            self.template_chooser.setEnabled(False)
         else:
             self.template_combo.setEnabled(False)
+
+        if self.search_on_disk_radio.isChecked():
             self.template_path.setEnabled(True)
             self.template_chooser.setEnabled(True)
+        else:
+            self.template_path.setEnabled(False)
+            self.template_chooser.setEnabled(False)
+
+    def toggle_no_map_radio(self, checked):
+        """Auto-connect slot when no map radio is clicked.
+
+        .. versionadded: 4.3.0
+        """
+        if checked:
+            self.button_open_composer.setEnabled(False)
+        else:
+            self.button_open_composer.setEnabled(True)
```

Comment:
I often write this too and then I noticed an easier way very late in my coding life ;-)
`self.button_open_composer.setEnabled(not checked)`

Reply:
ha!

Reply:
yeah, me too @Gustry